urbs.py

```diff
@@ -530,6 +533,12 @@ def create_model(data, timesteps=None, dt=1, dual=False):
         m.pro_tuples,
         rule=res_process_capacity_rule,
         doc='process.cap-lo <= total process capacity <= process.cap-up')
+    
+    m.res_roof_area = pyomo.Constraint(
+       m.pro_tuples,
```

Comment:
As discussed, this constraint should be declared over something like `m.pro_area_tuples`, a tuple set restricted to those site/process tuples that have a numeric `area-per-cap` attribute value.